tools/pythonpkg/src/numpy/type.cpp

```diff
@@ -10,93 +10,109 @@ static bool IsDateTime(const string &col_type_str) {
 	if (StringUtil::StartsWith(col_type_str, "datetime64[ns")) {
 		return true;
 	}
+	if (StringUtil::StartsWith(col_type_str, "datetime64[us")) {
+		return true;
+	}
+	if (StringUtil::StartsWith(col_type_str, "datetime64[ms")) {
+		return true;
+	}
+	if (StringUtil::StartsWith(col_type_str, "datetime64[s")) {
+		return true;
+	}
 	if (col_type_str == "<M8[ns]") {
 		return true;
 	}
 	return false;
 }
 
-PandasType ConvertPandasType(const py::object &col_type) {
+NumpyNullableType ConvertNumpyType(const py::handle &col_type) {
 	auto col_type_str = string(py::str(col_type));
 
+	// pandas.core.dtypes.base.StorageExtensionDtype
+	// this object also contains a 'numpy_dtype' attribute, which might be useful
+
+	if (StringUtil::EndsWith(col_type_str, "[pyarrow]")) {
+		col_type_str = col_type_str.substr(0, col_type_str.size() - string("[pyarrow]").size());
+	}
+
 	if (col_type_str == "bool" || col_type_str == "boolean") {
-		return PandasType::BOOL;
+		return NumpyNullableType::BOOL;
 	} else if (col_type_str == "uint8" || col_type_str == "UInt8") {
-		return PandasType::UINT_8;
+		return NumpyNullableType::UINT_8;
 	} else if (col_type_str == "uint16" || col_type_str == "UInt16") {
-		return PandasType::UINT_16;
+		return NumpyNullableType::UINT_16;
 	} else if (col_type_str == "uint32" || col_type_str == "UInt32") {
-		return PandasType::UINT_32;
+		return NumpyNullableType::UINT_32;
 	} else if (col_type_str == "uint64" || col_type_str == "UInt64") {
-		return PandasType::UINT_64;
+		return NumpyNullableType::UINT_64;
 	} else if (col_type_str == "int8" || col_type_str == "Int8") {
-		return PandasType::INT_8;
+		return NumpyNullableType::INT_8;
 	} else if (col_type_str == "int16" || col_type_str == "Int16") {
-		return PandasType::INT_16;
+		return NumpyNullableType::INT_16;
 	} else if (col_type_str == "int32" || col_type_str == "Int32") {
-		return PandasType::INT_32;
+		return NumpyNullableType::INT_32;
 	} else if (col_type_str == "int64" || col_type_str == "Int64") {
-		return PandasType::INT_64;
+		return NumpyNullableType::INT_64;
 	} else if (col_type_str == "float16" || col_type_str == "Float16") {
-		return PandasType::FLOAT_16;
+		return NumpyNullableType::FLOAT_16;
 	} else if (col_type_str == "float32" || col_type_str == "Float32") {
-		return PandasType::FLOAT_32;
-	} else if (col_type_str == "float64" || col_type_str == "Float64") {
-		return PandasType::FLOAT_64;
+		return NumpyNullableType::FLOAT_32;
+	} else if (col_type_str == "float64" || col_type_str == "Float64" || col_type_str == "double") {
```

Comment:
This is a type added by `pyarrow` and doesn't exist if the dtype backend is `numpy_nullable`, should probably be deleted for now as well